app/src/main/java/com/example/capstoneproject/ui/main/MainViewModel.kt

```diff
@@ -0,0 +1,52 @@
+package com.example.capstoneproject.ui.main
+
+import android.util.Log
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.capstoneproject.model.api.BitsoResponse
+import com.example.capstoneproject.model.api.Currency
+import com.example.capstoneproject.service.CurrencyService
+import kotlinx.coroutines.launch
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+class MainViewModel : ViewModel() {
+    private val retrofit = Retrofit.Builder()
+        .baseUrl("https://api.bitso.com/v3/")
+        .addConverterFactory(GsonConverterFactory.create())
+        .build()
+
+    private val service: CurrencyService = retrofit.create(CurrencyService::class.java)
```

Comment:
I suggest to move these lines to a new object file :)

Reply:
Could you explain a little bit more on this please? I don't get the comment